### PR TITLE
whitesur-gtk-theme: init at 2021-06-23

### LINKED ISSUE #127234: WhiteSur GTK Theme
**Project description**
Create a package for the WhiteSur GTK theme.

**Metadata**

* homepage URL: https://github.com/vinceliuice/WhiteSur-gtk-theme
* source URL: https://github.com/vinceliuice/WhiteSur-gtk-theme
* license: mit
* platforms: linux

By the same author as the Mojave GTK theme that there is already a package for: https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/themes/mojave/default.nix

This project uses a new build process though so I was unable to adapt the old package for the new theme.

Additionally the install script supports a large number of parameters to customize the theme. The readme details how this works. It would be nice if the package supported setting those parameters.

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

Add [WhiteSur-gtk-theme](https://github.com/vinceliuice/WhiteSur-gtk-theme).

Closes https://github.com/NixOS/nixpkgs/issues/127234

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/themes/whitesur/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    util-linux
+    which
+  ];
+
+  buildInputs = [
+    gnome-themes-extra # adwaita engine for Gtk2
+  ];
+
+  postPatch = ''
+    patchShebangs .
```
Comment (SuperSandro2000) [inline pkgs/data/themes/whitesur/default.nix:38]: Patching all files can take a while. Can we make this more precise?

File: pkgs/data/themes/whitesur/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "075fw57mv6zadq4dryn8bg2b3vq8inmisq18s758cv53pprxh9hw";
+  };
+
+  nativeBuildInputs = [
+    glib
+    libxml2
+    sassc
+    sudo
```
Comment (SuperSandro2000) [inline pkgs/data/themes/whitesur/default.nix:28]: Why sudo?

File: pkgs/data/themes/whitesur/default.nix (hunk 1 above)
Comment (romildo) [reply pkgs/data/themes/whitesur/default.nix]: Done.

File: pkgs/data/themes/whitesur/default.nix (hunk 2 above)
Comment (romildo) [reply pkgs/data/themes/whitesur/default.nix:28]: The installation script fails if `sudo` is not found.

https://github.com/vinceliuice/WhiteSur-gtk-theme/blob/master/lib-core.sh#L46

Although it is not used in our use case of the install script, I think.

Comment (SuperSandro2000) [reply pkgs/data/themes/whitesur/default.nix:28]: I would prefer your suggestion instead of providing sudo here.

Comment (romildo) [reply pkgs/data/themes/whitesur/default.nix]: Done.

**Pull request merged**